### PR TITLE
Feature - 2927 - Employment Equity Form

### LINKED ISSUE #2927: EE Data profile edit page
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description
Build the whole page.

## Screenshot: Prototype
[![a screenshot of ...][image]][image-link]
<sup>[Click screenshot to visit Figma.][image-link]</sup>

## Acceptance Criteria
- [ ] Each modal is its own form, which causes an API mutation when save is pressed
- [ ] EE data is not required, so this doesn't need to hook into isComplete status
- [ ] Page needs to load initial state, and show which selections are check, as well as pass state into modal forms
- [ ] In this case, Go Back button at bottom of page can simply be a link

   [image]: https://user-images.githubusercontent.com/3046459/171634039-d939b2f7-f97d-44d4-b128-74a8b6e4529d.png
   [image-link]: https://www.figma.com/proto/PrvXEDac0HVp7dF7SLWVFm/Candidate-profile-interfaces---V5.0.0?page-id=0%3A1&node-id=3139%3A4380&viewport=6038%2C-16292%2C0.78&scaling=min-zoom&starting-point-node-id=2340%3A4185
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

### PR DESCRIPTION
**Pull request by esizer**
Resolves #2927 

## Summary

This adds a component to the `/talent/profile/diversity-and-inclusion` page which allows users to add items to their profile. Before adding them, a modal with definition appears as a way to ensure users are reading definitions before self identifying. 

## Screenshot

Screen Shot 2022-06-06 at 4 12 33 PM

### CODE REVIEW

**Comment by vd1992** [comment]
should discuss that before any in-depth reviews, it is a good point you have made
not sure who in particular you should ask

**Comment by esizer** [comment]
> should discuss that before any in-depth reviews, it is a good point you have made
> not sure who in particular you should ask

Removed comments about complexity. This form is intended to have friction.

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusion.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import type { Meta, Story } from "@storybook/react";
+import { action } from "@storybook/addon-actions";
+import { fakeUsers } from "@common/fakeData";
+import { User } from "../../api/generated";
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusion.stories.tsx:5]: Unused import?

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.tsx [hunk 2]
```diff
@@ -27,15 +59,134 @@ export const DiversityEquityInclusionForm: React.FC = () => {
...
+        <li>
+          {intl.formatMessage({
+            defaultMessage:
+              "This data will be use in an anonymous form for statistical purposes.",
+            description:
+              "Explanation on how employment equity information will be used, item three",
+          })}
+        </li>
+      </ul>
+      <EquityOptions
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.tsx:145]: Seems like the bottom padding is lost in narrow viewports
Peek 2022-06-07 14-53
.

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/DialogActions.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+interface DialogActionsProps {
+  onDismiss: () => void;
+}
+
+const DialogActions: React.FC<DialogActionsProps> = ({ onDismiss }) => {
+  const intl = useIntl();
+  return (
+    <div
+      data-h2-display="b(flex)"
+      data-h2-flex-direction="b(column) m(row)"
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/DialogActions.tsx:15]: Seems like there's enough width here to show the buttons side by side.
image

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/VisibleMinorityDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import Dialog from "@common/components/Dialog";
+import { Checkbox } from "@common/components/form";
+
+import type { EquityDialogProps, EquityDialogFooterProps } from "../types";
+
+import DialogActions from "./DialogActions";
+
+// Question: Is this the correct URL?
+const statsCanLink = (...chunks: string[]) => (
+  <a href="https://www23.statcan.gc.ca/imdb/p3Var.pl?Function=DEC&Id=45152">
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/VisibleMinorityDialog.tsx:14]: @Jerryescandon could you please confirm the correct link?
Also, should this be a regular link or an "open in a new tab" link?

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/DisabilityDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+import Dialog from "@common/components/Dialog";
+import { Checkbox } from "@common/components/form";
+
+import type { EquityDialogProps, EquityDialogFooterProps } from "../types";
+
+import DialogActions from "./DialogActions";
+
+// Question: Is this the correct URL?
+const acaLink = (...chunks: string[]) => (
+  <a href="https://www.canada.ca/en/employment-social-development/programs/accessible-people-disabilities/act-summary.html#h2.02">
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/DisabilityDialog.tsx:14]: @Jerryescandon could you confirm the link, please?

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.test.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
+/**
+ * @jest-environment jsdom
+ */
+import "@testing-library/jest-dom";
+import { screen, fireEvent, waitFor } from "@testing-library/react";
+import React from "react";
+import { fakeUsers } from "@common/fakeData";
+import { act } from "react-dom/test-utils";
+import { UpdateUserAsUserInput } from "@common/api/generated";
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.test.tsx:9]: Unused ref?

**Review by petertgiles — CHANGES_REQUESTED** [review]
Wow, awesome!  A few small things but this looks great.

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusion.stories.tsx (hunk 1 above)
Comment (esizer) [reply frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusion.stories.tsx]: Man, I swear every PR I miss this. I need better glasses 👓 

Resolved with [remove unused imports](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2989/commits/19878f79437c9534a70d71917491db4691d7b662)

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.test.tsx (hunk 6 above)
Comment (esizer) [reply frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.test.tsx]: Resolved with [remove unused imports](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2989/commits/19878f79437c9534a70d71917491db4691d7b662)

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.tsx (hunk 2 above)
Comment (esizer) [reply frontend/talentsearch/src/js/components/diversityEquityInclusion/DiversityEquityInclusionForm.tsx:145]: Great catch! Resolved with [add margin to bottom of equity form](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2989/commits/d9fac380d8ab4d30530db1c1f3a4f2ac5efcd7f9)

File: frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/VisibleMinorityDialog.tsx (hunk 4 above)
Comment (esizer) [reply frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/VisibleMinorityDialog.tsx:14]: > Also, should this be a regular link or an "open in a new tab" link?

Just noting that if this is the expected behaviour we will also need to add a `(opens in new tab)` to the link text to meet WCAG 2.0.

_"For those not using assistive technology, the indication would also be available from the link text."_ - [WCAG 2.0 Techniques H83](https://www.w3.org/TR/WCAG20-TECHS/H83.html)

Comment (Jerryescandon) [reply frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/VisibleMinorityDialog.tsx:14]: Imma be real with y'all. In the next couple of days Imma have to make a new issue to update the copy on these modals. Which includes the 'href' which we're gonna link to. SO. this one seems correct for this version. BUT ALSO. I would definitely prefer it opens in a new tab.

Comment (petertgiles) [reply frontend/talentsearch/src/js/components/diversityEquityInclusion/dialogs/VisibleMinorityDialog.tsx:14]: Thanks Jerbo.  I'm going to consider "having the correct href" and "opening in a new tab" out of scope for now then.

**Review by petertgiles — APPROVED** [review]

**Pull request merged**